File: backend/app/src/main/java/com/yagubogu/talk/dto/v1/TalkEntranceResponse.java
package com.yagubogu.talk.dto.v1;

import com.yagubogu.game.domain.Game;
import com.yagubogu.member.domain.Member;

public record TalkEntranceResponse(
        String stadiumName,
        String homeTeamCode,
        String awayTeamCode,
        String myTeamCode
) {

    public static TalkEntranceResponse from(
            Game game,
            Member member
    ) {
        return new TalkEntranceResponse(
                game.getStadium().getFullName(),
                game.getHomeTeam().getTeamCode(),
                game.getAwayTeam().getTeamCode(),
                member.getTeam().getTeamCode()
        );
    }
}

File: mobile/composeApp/src/commonMain/kotlin/com/yagubogu/data/dto/response/talk/TalkEntranceResponse.kt
package com.yagubogu.data.dto.response.talk

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

// TODO 반영되면 수정해야 함
@Serializable
data class TalkEntranceResponse(
    @SerialName("stadiumName")
    val stadiumName: String, // 경기장 이름
    @SerialName("homeTeamCode")
    val homeTeamCode: String, // 홈팀 코드
    @SerialName("awayTeamCode")
    val awayTeamCode: String, // 어웨이팀 코드
    @SerialName("myTeamCode")
    val myTeamCode: String, // 내 팀 코드
)
